__test__/crud.test.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-undef */
+const request = require('supertest');
+const app = require('../server/app');
+const { SyllabusModel, connect, disconnect } = require('./db.mocks.js');
+const fixtures = require('./fixtures');
+
+beforeAll(() => {
+  connect();
+});
+
+afterEach(async () => {
+  await SyllabusModel.deleteMany();
+});
+
+afterAll(() => {
+  disconnect();
+});
+
+describe('Test the root path', () => {
+  test('It should respond to the GET method', async () => {
+    const response = await request(app).get('/');
+    expect(response.statusCode).toBe(200);
+  });
+
+  test('It should allow the insertion of a document', async () => {
+    let records = await SyllabusModel.find();
+    expect(records.length).toBe(0);
+    const response = await request(app).post('/api/syllabus', { body: fixtures.sampleSyllabusString });
```

Comment:
Suppose this is where the mocked db is getting tested right?

I'm wondering how can the "**request**" from **supertest** module know to to write data to the **localhost:27017/test** instead of the "real db" from the exposed env variables that is **localhost:27017/syllabuses**? Was there any config that you had to set up to make this happen?

I'm asking because when I looked in your **server/index.js** that has the that has all the CRUD routes from **app.js**, that server was connected to **localhost:27017/syllabuses** that was imported from your **db/index.js**, and now that **server/app.js** is imported here for testing. 

I might be having some wrong assumptions though so please correct me

Reply:
@phucciTr No you've got it! One of the interesting things about Mongo is that as long as _some_ mongo database is connected, all of the calls will use whatever the currently-connected instance is - including the database calls in `app.js`.
In production, the database comes from the `db` directory, where the actual mongo connection is set up and connected. This is what I needed to refactor last night before I could test since they needed to be dissociated. If you look in `server/index.js`, you can see that I'm calling the `connect()` function from the live database and then `app.listen()`. For testing, I'm using the mock version of the `connect()` method, which is listening on `localhost:27017/test`, and instead of `app.listen()`, I'm using `supertest` to mock the server functions based on what's in the `server/app.js` file.
In short, for production it's
- `connect()` to production database
- `app.listen()

For testing it's
- mock `connect()` to test database
- `request(app)` (`supertest` *instead* of `app.listen()`)